orchagent/txerrororch.cpp

```diff
@@ -0,0 +1,236 @@
+                            updateTimer(m_pollInterval);
+                        }
+                    }
+                    else if (fvField(fv) == "threshold")
+                    {
+                        int threshold = std::stoi(fvValue(fv));
+                        if(threshold<= 0)
+                        {
+                            SWSS_LOG_ERROR("Invalid threshold value, current threshold value %d", m_threshold);
+                        }
+                        else
+                        {
+                            m_threshold = threshold;
+                            SWSS_LOG_NOTICE("TX Error threshold set to %d", m_threshold);
+                        }
+                    }
+                }
+            }
+
+            //optional - fix If a del command is addad to the cli
+            else if (op == DEL_COMMAND)
+            {
+                m_pollInterval = DEFAULT_POLL_INTERVAL_SEC;
+                m_threshold = DEFAULT_THRESHOLD;
+            }
+
+        }
+        catch (const std::exception& e)
+        {
+            SWSS_LOG_ERROR("Exception was caught in the request parser in %s: %s", typeid(*this).name(), e.what());
+        }
+        catch (...)
+        {
+            SWSS_LOG_ERROR("Unknown exception was caught in the request parser");
+        }
+
+
+        if (erase_from_queue)
+        {
+            it = consumer.m_toSync.erase(it);
+        }
+        else
+        {
+            ++it;
+        }
+    }
+}
+
+void TxErrorOrch::updateTimer(int interval)
+{
+    SWSS_LOG_ENTER();
+
+    m_timer->stop();
+
+    auto interv = timespec { .tv_sec = interval, .tv_nsec = 0 };
+    m_timer->setInterval(interv);
+    m_timer->reset();
+    m_timer->start();
+}
+
+// Timer callback: check TX error counters for all ports
+void TxErrorOrch::doTask(swss::SelectableTimer &timer)
+{
+    SWSS_LOG_ENTER();
+
+    for (const auto& portPair : gPortsOrch->getAllPorts())
+    {
+        txErrorCounterCheck(portPair.second);
+    }
+}
+
+// Check TX error counter for a port and update status if threshold exceeded
+void TxErrorOrch::txErrorCounterCheck(const Port& port)
+{
+    SWSS_LOG_ENTER();
+
+    if(port.m_type != Port::PHY) 
+        return;
+
+    sai_object_id_t oid = port.m_port_id;
+
+    vector<FieldValueTuple> fieldValues;
+
+    if(!m_countersTable->get(sai_serialize_object_id(oid), fieldValues)){
+        SWSS_LOG_ERROR("Failed to get tx error counters for port %s", port.m_alias.c_str());
+        return;
+    }
+
+    uint64_t tx_errors = 0;
+    for(const auto& fv : fieldValues)
+    {
+        if(fvField(fv) =="SAI_PORT_STAT_IF_OUT_ERRORS")
+        {
+            tx_errors = std::stoull(fvValue(fv));
+            break;
+        }
+    }
+
+    if(tx_errors > static_cast<uint64_t>(m_threshold))
+    {
+        SWSS_LOG_NOTICE("TX Error counters for port %s exceeded threshold: %ld > %d", 
```

Comment:
this log message belongs to debug section - the user doesn't want to see each 10 seconds this log message for each port.
use SWSS_LOG_DEBUG